doc/dev_guide/release_management.rst

```diff
@@ -10,92 +10,166 @@ Release management
+Users will be informed about such critical patches via official Tarantool news
+channel (`tarantool_news <https://t.me/tarantool_news>`_).
 
-* **LTS** is a stable release which does not receive new features, and only gets
-  backward compatible fixes. Hence, following the rules of semver, LTS release
-  never has its MAJOR or MINOR version increased, and only gets PATCH level
-  releases.
+We also publish nightly builds, and use the fourth slot in the version
+identifier to designate the nightly build number.
 
-* **STABLE** is our current stable release, which may receive new features.
-  When the next STABLE version is published, MINOR version is incremented.
-  Between MINOR releases, we may have intermediate PATCH level releases as well,
-  which will contain only bug fixes. We maintain PATCH level releases for
-  two STABLE releases, the current and the previous one, to preserve support
-  continuity.
+.. _backward-incompatible:
 
-* **NEXT** is our next MAJOR release, and it follows the maturity
-  cycle described in the beginning. While NEXT release is in alpha state,
-  its MINOR is frozen at 0 and is only increased when the release reaches
-  BETA status. Once the NEXT release becomes STABLE, we switch the vehicle for
-  delivery of minor features, designating the previous stable release as LTS,
-  and releasing it with MINOR set to 10.
+.. note::
 
-To sum up, once a quarter we release:
+   A release series may contain backward incompatible changes in a sense that
+   existing Lua, SQL or C code that are run on a current release series
+   may not be run with the same effect on a future series.
+   However, we don't exploit this rule and don't make incompatible changes
+   without appropriate reason. We usually deliver information how mature
+   a functionality is via release notes.
 
-* the next LTS release, e.g. 2.10.6, 2.10.7 or 2.10.8
-* the next STABLE release, e.g. 3.6, 3.7 or 3.8
-* (optionally) an alpha or beta version of the NEXT release,
-  e.g. 4.0.1, 4.0.2 or 4.0.3
+   Please note that binary data layout
+   is always compatible with the previous series as well as with the LTS series
+   (an instance of ``X.Y`` version can be started on top of ``X.(Y+1)``
+   or ``1.10.z`` data); binary protocol is compatible too
+   (as client-server as well as replication protocol).
 
-In all supported releases, we also release a PATCH release as soon as we
-find and fix an outstanding CVE/vulnerability.
+.. _release-list:
 
-We also publish nightly builds, and use the fourth slot in the version
-identifier to designate the nightly build number.
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Release list
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Example version identifier:
+Below is the table containing all Tarantool releases starting from 1.10.0 up to
+the current latest versions. For each release series, releases are sorted out
```

Comment:
Maybe mention a date to don't be obligated to update the table at every release? Like so:

> up to the current latest versions (as on 2020-09-07).